client/my-sites/domains/domain-management/primary-domain/index.jsx

```diff
@@ -124,70 +128,103 @@ const PrimaryDomain = React.createClass( {
 								href={ primaryDomainSupportUrl }
 								target="_blank"
-								rel="noopener noreferrer"
-							>{ this.translate( 'Learn More' ) }</a>
+								rel="noopener noreferrer">
+								{ translate( 'Learn More' ) }
+							</a>
 						</div>
 					</section>
 
 					<Notice
 						showDismiss={ false }
 						className="primary-domain-notice">
-						{ this.translate(
-							'The primary domain for this site is currently ' +
-							'%(oldDomainName)s. If you update the primary ' +
-							'domain, all other domains will redirect to ' +
-							'%(newDomainName)s.',
-							{ args: {
-								oldDomainName: selectedSite.domain,
-								newDomainName: selectedDomainName
-							} }
-						) }
+						{
+							translate(
+								'The primary domain for this site is currently ' +
+								'%(oldDomainName)s. If you update the primary ' +
+								'domain, all other domains will redirect to ' +
+								'%(newDomainName)s.',
+								{ args: {
+									oldDomainName: selectedSite.domain,
+									newDomainName: selectedDomainName
+								} }
+							)
+						}
 					</Notice>
 
 					<section className="primary-domain__actions">
 						<button
 							className="button is-primary"
 							disabled={ this.state.loading }
 							onClick={ this.handleConfirmClick }>
-							{ this.translate( 'Update Primary Domain' ) }
+							{ translate( 'Update Primary Domain' ) }
 						</button>
 
 						<button
 							className="button"
 							disabled={ this.state.loading }
 							onClick={ this.handleCancelClick }>
-							{ this.translate( 'Cancel' ) }
+							{ translate( 'Cancel' ) }
 						</button>
 					</section>
 				</Card>
 			</Main>
 		);
 	}
-} );
-
-const mapStateToProps = state => {
-	const selectedSite = getSelectedSite( state );
-	const domains = getDomainsBySite( state, selectedSite );
-
-	return {
-		domains: {
-			isFetching: !! domains.length,
-			list: domains,
-		},
-		selectedSite,
-	};
-};
+}
+
+const cancelClick = ( { name, type } ) => composeAnalytics(
+	recordGoogleEvent(
+		'Domain Management',
+		`Clicked "Cancel" Button on a ${ type } in Primary Domain`,
+		'Domain Name',
+		name
+	),
+	recordTracksEvent(
+		'calypso_domain_management_primary_domain_cancel_click',
+		{ section: type }
+	),
+);
+
+const updatePrimaryDomainClick = ( { name, type }, success ) => composeAnalytics(
+	recordGoogleEvent(
+		'Domain Management',
+		`Clicked "Update Primary Domain" Button on a ${ type } in Primary Domain`,
+		'Domain Name',
+		name
+	),
+	recordTracksEvent(
+		'calypso_domain_management_primary_domain_update_primary_domain_click',
+		{
+			section: type,
+			success
+		}
+	),
+);
 
-const mapDispatchToProps = {
-	setPrimaryDomain: upgradesActions.setPrimaryDomain
-};
+export default connect(
+	state => {
+		const selectedSite = getSelectedSite( state );
+		const domains = getDomainsBySite( state, selectedSite );
 
-export default connect( mapStateToProps, mapDispatchToProps )( PrimaryDomain );
+		return {
+			domains: {
```

Comment:
An idea for a selector - `getSelectedDomain( selectedDomainName )`. No need to fetch the list and pass it as props (if the list changes, we'd re-render unnecessarily, etc.). Only `selectedDomainName` would be needed - the rest would be taken from the `state`, right? (heck, we could probably store that in the state too ;))